src/modules/git_status.rs

```diff
@@ -46,23 +47,35 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
                     "stashed" => info.get_stashed().and_then(|count| {
                         format_count(config.stashed, "git_status.stashed", count)
                     }),
-                    "ahead_behind" => info.get_ahead_behind().and_then(|(ahead, behind)| {
-                        if ahead > 0 && behind > 0 {
-                            format_text(config.diverged, "git_status.diverged", |variable| {
-                                match variable {
-                                    "ahead_count" => Some(ahead.to_string()),
-                                    "behind_count" => Some(behind.to_string()),
-                                    _ => None,
+                    "ahead_behind" => {
+                        info.get_ahead_behind()
+                            .and_then(|(ahead, behind)| match (ahead, behind) {
+                                (None, _) | (_, None) => None,
+                                (Some(a), Some(b)) => {
```

Comment:
I think you can keep the changes smaller and simplify this by using the `?` operator:

```rust
match (ahead?, behind?) {
```